contrib/repro_test.bzl

```diff
@@ -0,0 +1,243 @@
+    # Note that this target must always be relative to the project's root.
+    test_image = "//tests/container:set_cmd",
+    base = "@bazel_latest//image",
+    src_project_files = "//:src_project",
+    container_diff_args: [
+        "--type=history",
+        "--type=file",
+        "--type=size",
+        "--type=rpm",
+        "--type=apt",
+        "--type=pip",
+        "--type=node",
+    ],
+)
+"""
+
+load("@base_images_docker//util:run.bzl", _extract = "extract")
+load("@bazel_skylib//lib:dicts.bzl", "dicts")
+load("//container:container.bzl", _container = "container")
+load("//skylib:filetype.bzl", container_filetype = "container")
+
+def _impl(ctx):
+    """Core implementation of container_repro_test"""
+
+    # Check for existence of the WORKSPACE file in the files provided.
+    src_project_files = ctx.files.src_project_files
+    workspace_file = None
+    for f in src_project_files:
+        if f.basename == "WORKSPACE":
+            workspace_file = f
+            break
+
+    if not workspace_file:
+        fail("A WORKSPACE file must be among the src_project_files provided files.")
+
+    name = ctx.attr.name
+
+    # Record the absolute path to the provided WORKSPACE file and assume it to
+    # be the root of the Bazel project to build the test_image in.
+    proj_root = ctx.actions.declare_file(name + "_src_project_root.txt")
+    ctx.actions.run_shell(
+        command = "readlink -f %s | xargs dirname >> %s" % (workspace_file.path, proj_root.path),
+        inputs = [workspace_file],
+        outputs = [proj_root],
+    )
+
+    src_paths = [f.path for f in src_project_files]
+    proj_tar = ctx.actions.declare_file(name + "_src_project.tar")
+
+    # Tar the provided source project files to mount into the image that will
+    # be used to build and repro the test_image in.
+    ctx.actions.run_shell(
+        command = "readlink -f %s | xargs tar -cvf %s" % (" ".join(src_paths), proj_tar.path),
+        inputs = src_project_files,
+        outputs = [proj_tar],
+    )
+
+    image_tar = ctx.actions.declare_file(name + ".tar")
+
+    # Build the Bazel image to build and repro the test_image in.
+    # Mount the source project and its path into this image.
+    _container.image.implementation(
+        ctx,
+        files = [proj_root],
+        tars = [proj_tar],
+        output_executable = ctx.actions.declare_file(name + "_load.sh"),
+        output_tarball = image_tar,
+        workdir = "/",
+    )
+
+    test_img_label = ctx.attr.test_image.label
+    build_targets = [
+        str(test_img_label),
+        str(test_img_label) + ".tar",
+        str(test_img_label) + ".digest",
+        str(test_img_label) + ".json",
+    ]
+    extract_path = "/img_outs"
+    commands = [
+        "cd \$(cat /%s)" % proj_root.basename,
+        "bazel build " + " ".join(build_targets),
+        "cp -r bazel-bin/%s %s" % (test_img_label.package, extract_path),
+    ]
+    docker_run_flags = ["--entrypoint", "''"]
```

Comment:
you need to pass `-v /var/run/docker.sock:/var/run/docker.sock` to enable sibling docker
if the image being built is using download_pkgs or install_pkgs (which toolchain_container does indirectly) the build process will fail

Reply:
Done